spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ImplicitExpressionConfiguration.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.boot.configurationmetadata.ConfigurationMetadataRepositoryJsonBuilder;
+import org.springframework.cloud.stream.annotationprocessor.ImplicitExpressionProcessor;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.PriorityOrdered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A configuration that installs a low level {@link org.springframework.core.env.PropertySource}
+ * to support {@link org.springframework.cloud.stream.annotationprocessor.ImplicitExpression} properties.
+ *
+ * @author Eric Bottard
+ * @author Florent Biville
+ */
+@Configuration
+public class ImplicitExpressionConfiguration implements EnvironmentAware, PriorityOrdered {
+
+
+	private Map<String, ConfigurationMetadataProperty> allProperties;
+
+	public ImplicitExpressionConfiguration() throws IOException {
+		ConfigurationMetadataRepositoryJsonBuilder builder = ConfigurationMetadataRepositoryJsonBuilder.create();
+		ResourcePatternResolver moduleResourceLoader = new PathMatchingResourcePatternResolver();
+		for (Resource r : moduleResourceLoader.getResources("classpath*:" + ImplicitExpressionProcessor.METADATA_FILE)) {
+			builder.withJsonResource(r.getInputStream());
+		}
+		allProperties = builder.build().getAllProperties();
+	}
+
+	@Override
+	public void setEnvironment(final Environment environment) {
+		if (environment instanceof ConfigurableEnvironment) {
+			ConfigurableEnvironment configurableEnvironment = (ConfigurableEnvironment) environment;
+
+			Map<String, Object> map = new HashMap<>();
+			MapPropertySource fakeValues = new MapPropertySource("Implicit expressions", map);
+			for (Map.Entry<String, ConfigurationMetadataProperty> entry : allProperties.entrySet()) {
+				String raw = environment.getProperty(entry.getKey());
+				if (raw != null) {
+					String type = entry.getValue().getType();
+					map.put(entry.getKey() + "-expression", literalExpression(raw, type));
+				}
+			}
+			configurableEnvironment.getPropertySources().addLast(fakeValues);
+		}
+	}
+
+	/**
+	 * Return a literal SpEL expression with the given value, using proper literal notation according to the wanted
+	 * type.
+	 */
+	private String literalExpression(Object value, String type) {
+		Class<?> clazz = ClassUtils.resolveClassName(type, Thread.currentThread().getContextClassLoader());
+		if (CharSequence.class.isAssignableFrom(clazz)) {
+			return String.format("'%s'", value);
+		}
+		else if (Number.class.isAssignableFrom(clazz)) {

```

Comment:
That's the question really. I think we'll see in the wild, but thinking about the modules we've already authored, I don't many more. Maybe arrays of the above.